weld/llvm.rs

```diff
@@ -1993,18 +1993,83 @@ impl LlvmGenerator {
+                                        let iden_elem = match *op {
+                                            BinOpKind::Add => {
+                                                match **elem_ty {
+                                                    Scalar(F32) | Scalar(F64) => "0.0".to_string(),
+                                                    _ => "0".to_string(),
+                                                }
+                                            }
+                                            BinOpKind::Multiply => {
+                                                match **elem_ty {
+                                                    Scalar(F32) | Scalar(F64) => "1.0".to_string(),
+                                                    _ => "1".to_string(),
+                                                }
+                                            }
+
+                                            _ => {
+                                                return weld_err!("Invalid merger binary op in \
+                                                                  codegen")
+                                            }
+                                        };
+
+                                        let init_elem = match *arg {
+                                            Some(ref s) => {
+                                                let arg_str = self.load_var(llvm_symbol(s).as_str(),
+                                                              &elem_type,
+                                                              ctx)?;
+                                                arg_str
+                                            }
+                                            _ => iden_elem.clone(),
+                                        };
+
+                                        let first = ctx.var_ids.next();
+                                        let nworkers = ctx.var_ids.next();
+                                        let i = ctx.var_ids.next();
+                                        let cur_ptr = ctx.var_ids.next();
+                                        let i2 = ctx.var_ids.next();
+                                        let cond = ctx.var_ids.next();
+                                        let cond2 = ctx.var_ids.next();
+
+                                        // Generate label names.
+                                        let label_base = ctx.var_ids.next();
+                                        let mut label_ids =
+                                            IdGenerator::new(&label_base.replace("%", ""));
+                                        let entry = label_ids.next();
+                                        let body = label_ids.next();
+                                        let done = label_ids.next();
+
+
+                                        ctx.code.add(format!(include_str!("resources/merger/init_merger.ll"),
+                                        first=first,
```

Comment:
Yup, it is; `rustfmt` actually throws some error on this line because it's too long.